code/modules/projectiles/projectile/ego_bullets/aleph.dm

```diff
@@ -37,3 +37,68 @@
 		L.apply_damage(50, BLACK_DAMAGE, null, L.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
 	return BULLET_ACT_HIT
 
+/obj/projectile/ego_bullet/nihil
+	name = "dark energy"
+	icon_state = "nihil"
+	desc = "Just looking at it seems to suck the life out of you..."
+	damage = 40	//Fires 3 +10 damage per upgrade, up to 80
+	speed = 0.8
+	damage_type = WHITE_DAMAGE
+	flag = WHITE_DAMAGE
+	hitsound = 'sound/abnormalities/nihil/filter.ogg'
+	var/damage_list = list(WHITE_DAMAGE)
+	var/icon_list = list()
+	var/wrath
+	var/despair
+	var/greed
+	var/hate
+
+/obj/projectile/ego_bullet/nihil/on_hit(atom/target, blocked = FALSE)
+	if(ishuman(target) && isliving(firer) && hate) //this only happens with the queen of hatred upgrade
+		var/mob/living/carbon/human/H = target
+		var/mob/living/user = firer
+		if(firer==target)
+			return BULLET_ACT_BLOCK
+		if(user.faction_check_mob(H)) // Our faction
+			if(H.is_working)
+				H.visible_message("<span class='warning'>[src] vanishes on contact with [H]... but nothing happens!</span>")
+				qdel(src)
+				return BULLET_ACT_BLOCK
+			switch(damage_type)
+				if(WHITE_DAMAGE)
+					H.adjustSanityLoss(-damage*0.2)
+				if(BLACK_DAMAGE)
+					H.adjustBruteLoss(-damage*0.1)
+					H.adjustSanityLoss(-damage*0.1)
+				else // Red or pale
+					H.adjustBruteLoss(-damage*0.2)
+			H.visible_message("<span class='warning'>[src] vanishes on contact with [H]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	..()
```

Comment:
Probably should invert the conditionals to allow it to return `..()` early, allowing the rest of the code to be a level lower and easier to read.